go/cmd/dolt/commands/cnfcmds/auto_resolve.go

```diff
@@ -63,26 +62,77 @@ func AutoResolveAll(ctx context.Context, dEnv *env.DoltEnv, strategy AutoResolve
 // AutoResolveTables resolves all conflicts in the given tables according to the
 // given |strategy|.
 func AutoResolveTables(ctx context.Context, dEnv *env.DoltEnv, strategy AutoResolveStrategy, tbls []string) error {
-	root, err := dEnv.WorkingRoot(ctx)
+	ws, err := dEnv.WorkingSet(ctx)
 	if err != nil {
 		return err
 	}
 
-	for _, tblName := range tbls {
-		err = ResolveTable(ctx, dEnv, root, tblName, strategy)
+	// schema conflicts
+	if ws.MergeActive() {
+		ws, err = ResolveSchemaConflicts(ctx, dEnv.DoltDB, ws, tbls, strategy)
 		if err != nil {
 			return err
 		}
+		if err = dEnv.UpdateWorkingSet(ctx, ws); err != nil {
+			return err
+		}
 	}
 
+	// data conflicts
+	for _, tblName := range tbls {
+		err = ResolveDataConflicts(ctx, dEnv, ws, tblName, strategy)
+		if err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
-// ResolveTable resolves all conflicts in the given table according to the given
+func ResolveSchemaConflicts(ctx context.Context, ddb *doltdb.DoltDB, ws *doltdb.WorkingSet, tables []string, strategy AutoResolveStrategy) (*doltdb.WorkingSet, error) {
```

Comment:
(minor) might be worth putting this code into a shared package, if there's an easy one that won't cause an import cycle. It looked pretty compatible with the version in `dprocedures/dolt_conflicts_resolve.go`, but I didn't check every detail.

Reply:
yeah I think we should, will follow up with this